src/SonarScanner.MSBuild.Shim/PropertiesFileGenerator.cs

```diff
@@ -110,31 +110,36 @@ public ProjectInfoAnalysisResult GenerateFile()
             return result;
         }
 
-        public bool TryWriteProperties(PropertiesWriter writer, out IEnumerable<ProjectData> allProjects)
-        {
-            var projects = ProjectLoader.LoadFrom(analysisConfig.SonarOutputDir);
+        public bool TryWriteProperties(PropertiesWriter writer, out IEnumerable<ProjectData> allProjects) =>
+            TryWriteProperties(writer, ProjectLoader.LoadFrom(analysisConfig.SonarOutputDir).ToArray(), out allProjects);
 
-            if (!projects.Any())
+        public bool TryWriteProperties(PropertiesWriter writer, IList<ProjectInfo> projects, out IEnumerable<ProjectData> allProjects)
+        {
+            if (projects.Count == 0)
             {
                 logger.LogError(Resources.ERR_NoProjectInfoFilesFound, SonarProduct.GetSonarProductToLog(analysisConfig.SonarQubeHostUrl));
-                allProjects = Enumerable.Empty<ProjectData>();
+                allProjects = [];
                 return false;
             }
 
-            var projectDirectories = projects.Select(p => p.GetDirectory()).ToList();
             var analysisProperties = analysisConfig.ToAnalysisProperties(logger);
             FixSarifAndEncoding(projects, analysisProperties);
-            allProjects = projects.GroupBy(p => p.ProjectGuid).Select(ToProjectData).ToList();
-            var validProjects = allProjects.Where(p => p.Status == ProjectInfoValidity.Valid).ToList();
-
+            allProjects = projects.GroupBy(x => x.ProjectGuid).Select(ToProjectData).ToList();
+            var validProjects = allProjects.Where(x => x.Status == ProjectInfoValidity.Valid).ToList();
             if (validProjects.Count == 0)
             {
                 logger.LogError(Resources.ERR_NoValidProjectInfoFiles, SonarProduct.GetSonarProductToLog(analysisConfig.SonarQubeHostUrl));
                 return false;
             }
 
+            var projectDirectories = validProjects.Select(x => x.Project.GetDirectory()).ToList();
```

Comment:
We no longer consider invalid projects (temp/generated projects are considered invalid) when the common path is computed.